File: app/src/main/java/de/garten/training/depotflow/data/mapper/OutboxMapper.java
package de.garten.training.depotflow.data.mapper;

import de.garten.training.depotflow.data.api.dto.OutboxDto;
import de.garten.training.depotflow.data.db.green.SyncOutboxEntry;

public class OutboxMapper {

    public OutboxDto toDto(SyncOutboxEntry entry) {
        return new OutboxDto(
                entry.getAggregateType(),
                entry.getAggregateId(),
                entry.getOperation(),
                entry.getPayload(),
                entry.getCreatedAt()
        );
    }
}

File: app/src/main/java/de/garten/training/depotflow/data/mapper/OutboxMapper.kt
package de.garten.training.depotflow.data.mapper

import de.garten.training.depotflow.data.api.dto.OutboxDto
import de.garten.training.depotflow.data.db.green.SyncOutboxEntry

class OutboxMapper {
    fun toDto(entry: SyncOutboxEntry): OutboxDto {
        return OutboxDto(
            entry.aggregateType,
            entry.aggregateId,
            entry.operation,
            entry.payload,
            entry.createdAt
        )
    }
}
